gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsService.java

```diff
@@ -0,0 +1,508 @@
+    }
+  }
+
+  /**
+   * Class for specifying zone listing options.
+   */
+  class ZoneListOptions extends AbstractOption implements Serializable {
+
+    private static final long serialVersionUID = -7922038132321229290L;
+
+    ZoneListOptions(DnsServiceRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the zones's fields to be returned by the RPC call.
+     *
+     * <p>If this option is not provided all zone fields are returned. {@code
+     * ZoneFieldOptions.fields} can be used to specify only the fields of interest. Zone ID is
+     * always returned, even if not specified. {@link ZoneField} provides a list of fields that can
+     * be used.
+     */
+    public static ZoneListOptions fields(ZoneField... fields) {
+      return new ZoneListOptions(DnsServiceRpc.Option.FIELDS, ZoneField.selector(fields));
+    }
+
+    /**
+     * Returns an option to specify a page token.
+     *
+     * <p>The page token (returned from a previous call to list) indicates from where listing should
+     * continue.
+     */
+    public static ZoneListOptions pageToken(String pageToken) {
+      return new ZoneListOptions(DnsServiceRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * The maximum number of zones to return per RPC.
+     *
+     * <p>The server can return fewer zones than requested. When there are more results than the
+     * page size, the server will return a page token that can be used to fetch other results.
+     */
+    public static ZoneListOptions pageSize(int pageSize) {
+      return new ZoneListOptions(DnsServiceRpc.Option.PAGE_SIZE, pageSize);
+    }
+
+    /**
+     * Restricts the list to return only zones with this fully qualified domain name.
+     */
+    public static ZoneListOptions dnsName(String dnsName) {
+      return new ZoneListOptions(DnsServiceRpc.Option.DNS_NAME, dnsName);
+    }
+  }
+
+  /**
+   * Class for specifying project options.
+   */
+  class ProjectOptions extends AbstractOption implements Serializable {
+
+    private static final long serialVersionUID = 6817937338218847748L;
+
+    ProjectOptions(DnsServiceRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the project's fields to be returned by the RPC call.
+     *
+     * <p>If this option is not provided all project fields are returned. {@code
+     * ProjectOptions.fields} can be used to specify only the fields of interest. Project ID is
+     * always returned, even if not specified. {@link ProjectField} provides a list of fields that
+     * can be used.
+     */
+    public static ProjectOptions fields(ProjectField... fields) {
+      return new ProjectOptions(DnsServiceRpc.Option.FIELDS, ProjectField.selector(fields));
+    }
+  }
+
+  /**
+   * Class for specifying change request field options.
+   */
+  class ChangeRequestFieldOptions extends AbstractOption implements Serializable {
```

Comment:
ChangeRequestOption or specific ones ChangeRequestGetOption and ChangeRequestCreateOption